src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.CSharpCodeGenerator.ExpressionCodeGenerator.cs

```diff
@@ -22,15 +24,16 @@ internal partial class CSharpMethodExtractor
     {
         private partial class CSharpCodeGenerator
         {
-            private class ExpressionCodeGenerator : CSharpCodeGenerator
+            private sealed class ExpressionCodeGenerator : CSharpCodeGenerator
             {
                 public ExpressionCodeGenerator(
                     InsertionPoint insertionPoint,
                     SelectionResult selectionResult,
                     AnalyzerResult analyzerResult,
-                    OptionSet options,
+                    CSharpCodeGenerationOptions options,
+                    NamingStylePreferencesProvider namingPreferences,
```

Comment:
Naming preferences are separate from code gen options since they are fairly expensive and not always needed. Currently they are also not messagepack-serializable. Will follow up on that.